torch/lib/c10d/ProcessGroupMPI.cpp

```diff
@@ -717,7 +718,7 @@ c10::intrusive_ptr<ProcessGroup::Work> ProcessGroupMPI::alltoall(
         at::Tensor dstFlatData = at::empty({dst_len}, dstdata[0].options());
         auto srcFlatDataSplits =
             srcFlatData.split_with_sizes(c10::IntArrayRef(send_lengthsL), 0);
-        for (int i = 0; i < size_; i++) {
+        for(const auto i : c10::irange(size_)) {
```

Comment:
Curious to know what is the benefit of using irange instead of regular loops? Is this mostly for readability?

Reply:
I've answered more fully via DM. In short: type safety, const safety, and consistency.